src/components/form-editor/field-editor/FieldEditor.tsx

```diff
@@ -0,0 +1,352 @@
+import FormRow from "components/common/form-row";
+import FieldVariant from "constants/FieldVariant";
+import QuestionType from "constants/QuestionType";
+import { DynamicFieldsContext } from "context/DynamicFieldsContext";
+import { DefaultField, DynamicField } from "types";
+
+import QuestionTypeLabel from "../question-type-label";
+import useStyles from "./styles";
+
+// unique identifier for options
+let OPTION_KEY_COUNTER = 1;
+
+const generateKey = (): number => {
+  const key = OPTION_KEY_COUNTER;
+  OPTION_KEY_COUNTER += 1;
+  return key;
+};
+
+const getSelectIndicator = (className: string, questionType: string) =>
+  questionType === QuestionType.MULTIPLE_SELECT ? (
+    <Checkbox className={className} disabled size="small" />
+  ) : (
+    <Radio className={className} disabled size="small" />
+  );
+
+type OptionFormData = {
+  index: number;
+  value: string;
+};
+
+type FieldFormData = Omit<DefaultField | DynamicField, "options"> & {
+  options: OptionFormData[];
+};
+
+type Props = {
+  field: DefaultField | DynamicField;
+  isDefault: boolean;
+  isEnabled?: boolean;
+  isReadOnly: boolean;
+  onChangeEnabled?: (enabled: boolean) => void;
+};
+
+const FieldEditor = ({
+  field,
+  isDefault,
+  isEnabled = true,
+  isReadOnly,
+  onChangeEnabled = () => {},
+}: Props) => {
+  const { fetchDynamicFields } = useContext(DynamicFieldsContext);
+  const classes = useStyles({ isDefault });
+  const [isEditing, setIsEditing] = useState(false);
+  const [fieldFormData, setFieldFormData] = useState<FieldFormData>({
+    ...field,
+    options: [],
+  });
+  const [showEditConfirmationDialog, setShowEditConfirmationDialog] = useState(
+    false
+  );
+  const [
+    showDeleteConfirmationDialog,
+    setShowDeleteConfirmationDialog,
+  ] = useState(false);
+
+  useEffect(() => {
+    setFieldFormData({
+      ...field,
+      options: field.options.map((option) => ({
+        index: generateKey(),
+        value: option,
+      })),
+    });
+  }, [field]);
+
+  const onAddOption = (): void => {
+    setFieldFormData({
+      ...fieldFormData,
+      options: [...fieldFormData.options, { index: generateKey(), value: "" }],
+    });
+  };
+
+  const onUpdateOption = (data: OptionFormData): void => {
+    const options = [...fieldFormData.options];
+    const optionIndex = options.findIndex(
+      (option) => option.index === data.index
+    );
+    options[optionIndex] = data;
+    setFieldFormData({ ...fieldFormData, options });
+  };
+
+  const onDeleteOption = (index: number): void => {
+    setFieldFormData({
+      ...fieldFormData,
+      options: fieldFormData.options.filter((option) => option.index !== index),
+    });
+  };
+
+  const onSubmitField = async () => {
+    const data = {
+      ...field,
+      id: Number(field.id),
+      options: fieldFormData.options.map((option) => option.value),
+    };
+    try {
+      await DynamicFieldAPI.putField(data);
```

Comment:
this endpoint doesn't actually exist in the backend yet, so will throw a 404 until it's done